web/src/firebase/firebasefunctions.tsx

```diff
@@ -0,0 +1,50 @@
+// firebasefunctions.tsx
+
+import { db } from './firebaseconfig';
+import { collection, query, where, getDocs, setDoc, doc } from 'firebase/firestore';
+import axios from 'axios';
+
+// Utility function to check if the show exists in Firestore
+async function checkShowExists(showName: string): Promise<{ exists: boolean, data?: any }> {
+  const showsRef = collection(db, 'shows');
```

Comment:
I think you wrote the functions for cloud firestore but we are using realtime database